carball/analysis/stats/boost/boost.py

```diff
@@ -33,12 +36,25 @@ def calculate_player_stat(self, player_stat_map: Dict[str, PlayerStats], game: G
             proto_boost.time_low_boost = self.get_time_with_low_boost(data_frame, player_data_frame)
             proto_boost.time_no_boost = self.get_time_with_zero_boost(data_frame, player_data_frame)
 
+            proto_boost.num_stolen_boosts = self.get_num_stolen_boosts(player_data_frame,
+                                                                       player_map[player_key].is_orange)
+
     @staticmethod
     def get_player_boost_usage(player_dataframe: pd.DataFrame) -> np.float64:
         _diff = -player_dataframe.boost.diff()
         boost_usage = _diff[_diff > 0].sum() / 255 * 100
         return boost_usage
 
+    @classmethod
+    def get_num_stolen_boosts(cls, player_dataframe: pd.DataFrame, is_orange):
+        filtered = player_dataframe[player_dataframe.boost_collect == True]
```

Comment:
To avoid having a variable named a generic `filtered`,

I'd do 
`
boost_collected_dataframe = player_dataframe[player_dataframe.boost_collect == True]
boost_collected_in_opposing_third_dataframe = cls.field_constants.get_third_2(boost_collected_dataframe) if is_orange == 0 else cls.field_constants.get_third_0(boost_collected_dataframe)
return len(boost_collected_in_opposing_third_dataframe)
`